source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -108,8 +121,16 @@ Http::FilterHeadersStatus CacheFilter::encodeHeaders(Http::ResponseHeaderMap& he
     insert_ = cache_.makeInsertContext(std::move(lookup_), *encoder_callbacks_);
     // Add metadata associated with the cached response. Right now this is only response_time;
     const ResponseMetadata metadata = {time_source_.systemTime()};
+    // TODO (capoferro): Note that there is currently no way to communicate back to the CacheFilter
+    // that an insert has failed. This is something we want to fix in future.
```

Comment:
why?

A user of a cache should never assume that an insert is prevented from being evicted for any length of time, right? So why is it necessary to feed back insertion status? What decision can be made from that?

Reply:
For larger objects that are streamed to the cache, if an insert fails partway, it's better not to send additional chunks to the cache if we're already in a failure state and should abort. Additionally, it's useful for metrics gathering to know more detail on failed insertions, and I am thinking metrics should be gathered at the filter level rather than passing metrics objects down into the HttpCache.

Reply:
Good point -- put this PR comment in as a code comment?